lib/IDE/SourceEntityWalker.cpp

```diff
@@ -634,15 +634,22 @@ ASTWalker::PreWalkAction SemaAnnotator::walkToTypeReprPre(TypeRepr *T) {
       return Action::StopIf(!Continue);
     }
   } else if (auto FT = dyn_cast<FixedTypeRepr>(T)) {
-    ValueDecl *VD = FT->getType()->getAnyGeneric();
-    if (auto DT = FT->getType()->getAs<DynamicSelfType>())
+    if (ValueDecl *VD = FT->getType()->getAnyGeneric()) {
+      auto Data = ReferenceMetaData(SemaReferenceKind::TypeRef, None);
+      auto Continue = passReference(VD, FT->getType(), FT->getLoc(),
+                                    FT->getSourceRange(), Data);
+      return Action::StopIf(!Continue);
```

Comment:
Now that SelfTypeRepr has been separated out, this case should only handle implicit cases, so I think it would make sense to set both `isImplicit` and `isImplicitCtorType` to `true` on the `ReferenceMetaData`

Reply:
setting `isImplicitCtorType` changes this case: https://github.com/apple/swift/pull/65729/files#diff-1563554316b92e9e77632af5f813d4a441724dfd389f42d401ff291ee254557cR34 to be implicit. If it weren't for the ambiguous type it wouldn't be, so I'm not sure it should be here either?

Reply:
ah good point, done!